coordinator/cluster/util.go

```diff
@@ -8,54 +8,54 @@ import (
 )
 
 type sortableAgent struct {
-	order int
-	Users int
+	index int
+	users int
 }
 
 // gives numbers to distribute users evenly
 func populateSortableAgents(agents []*agent.LoadAgent) ([]sortableAgent, error) {
-	sortedAgents := make([]sortableAgent, 0)
 	if len(agents) == 0 {
 		return nil, errors.New("input slice length must be greater than 0")
 	}
+	sortableAgents := make([]sortableAgent, len(agents))
 	for i, a := range agents {
-		sortedAgents = append(sortedAgents, sortableAgent{
-			order: i,
-			Users: a.Status().NumUsers,
-		})
+		sortableAgents[i] = sortableAgent{
+			index: i,
+			users: a.Status().NumUsers,
+		}
```

Comment:
This is me being *pedantic* but since the allocated array is of values we don't need to create anything new but can just assign.

```go
sortableAgents[i].index = i
sortableAgents[i].users = a.Status().NumUsers
```

Reply:
No problem, good point 👍